lib/facility.rb

```diff
@@ -1,14 +1,69 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_data)
+    @name = facility_data[:name]
+    @address = facility_data[:address]
+    @phone = facility_data[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  end
+
+  def add_service(service)
+    unless @services.include?(service)
+      @services << service
+    end
+  end
```

Comment:
Nice! I like this edge case to check if it's included first before adding the service.